packages/trackerless-network/src/logic/neighbor-discovery/NeighborUpdateRpcLocal.ts

```diff
@@ -44,19 +58,9 @@ export class NeighborUpdateRpcLocal implements INeighborUpdateRpc {
                 ))
             )
             this.config.neighborFinder.start()
-            const response: NeighborUpdate = {
-                streamPartId: this.config.streamPartId,
-                neighborDescriptors: this.config.neighbors.getAll().map((neighbor) => neighbor.getPeerDescriptor()),
-                removeMe: false
-            }
-            return response
+            return this.createResponse(false)
         } else {
-            const response: NeighborUpdate = {
-                streamPartId: this.config.streamPartId,
-                neighborDescriptors: this.config.neighbors.getAll().map((neighbor) => neighbor.getPeerDescriptor()),
-                removeMe: true
-            }
-            return response
+            return this.createResponse(true)
```

Comment:
Could simplify the if so that there are only two if branches, one which returns `this.createResponse(true)` and another which returns `this.createResponse(false)`

Reply:
Played around with copilot "simplify" feature and it suggested the following, looks kinda good? What did you have in mind?

```ts
if (!this.config.neighbors.has(senderId)) {
    return this.createResponse(true)
} else {
    const isOverNeighborCount = this.config.neighbors.size() > this.config.neighborCount
      && message.neighborDescriptors.length > this.config.neighborCount

    if (!isOverNeighborCount) {
     // do all the updating stuff here
        this.config.neighborFinder.start()
    }

    return this.createResponse(isOverNeighborCount)
}
```

Reply:
Now we always call this.updateContacts. Is this intentional? Before commit 92d01df75df83aa4ead54a10c67fe974db3d3792 this logic was only in `else if (this.config.neighbors.has(senderId))` branch.

Reply:
Yes this is intentional. The contacts can be updated regardless of weather or not the neighbor is asked to disconnect